### ISSUE TITLE
Query: function `increase` calculates incorrect increase

### ISSUE THREAD

**Issue by Maros112358** [grafana/grafana#78053]
### What happened?

I have a metric with value increasing linearly with time. Approximately every 24hours its value decreases rapidly. My graph then looks like a see-saw shown below. The query is like
```
my_metric{key='value'}
```
<img width="1883" alt="image" src="https://github.com/grafana/grafana/assets/98359168/9eeb440a-f1ee-4473-963b-e63b08bc1b5f">

I would like to calculate how much the value increased during last hour. I want to get only **non-negative** value changes. I can't use `delta`, because it gives me negative values when the value drops. I decided to use `increase`, because from documentation I understood, that it would give me **non-negative** changes in the values. Using the following query gives me the graph below
```
increase(my_metric{key='value'}[1h])
```
<img width="1885" alt="image" src="https://github.com/grafana/grafana/assets/98359168/8b6a7b9e-5b39-4240-8fcd-ca103e347cab">

### What did you expect to happen?

I expected the data to be without the peaks, because those big positive peaks do not appear when using `delta`.

<img width="1883" alt="image" src="https://github.com/grafana/grafana/assets/98359168/262140ad-de68-47df-8851-ae55d986bd2b">

The `increase` function without peaks should look like this
<img width="1884" alt="image" src="https://github.com/grafana/grafana/assets/98359168/088383cc-6060-4752-ba5f-7a9c43d97b76">

### Did this work before?

No.

### How do we reproduce it?

1. Take a zig-saw data
2. Calculate `increase` of the data during last hour
3. Observe

### Environment (with versions)?

Grafana: Grafana v10.3.0-63137
OS: MacOS
Browser: Google Chrome

### Grafana platform?

I use Grafana Cloud

### Datasource(s)?

Prometheus

**Comment by kylebrandt** [comment]
For reference:

https://prometheus.io/docs/prometheus/latest/querying/functions/#increase

The results also don't line up how I read the docs, but not an expert, we will ask our prom experts.

**Comment by itsmylife** [comment]
`delta` calculates the difference between the first and last value. It should only be used with gauge metrics.
`increase`, on the other hand, does the same but with one difference. Before calculation, a `reset correction` is being done. Please see this video https://youtu.be/7uy_yovtyqw?t=156 `increase` should only be used with counter metrics.
So the graph you expected to is not possible for `increase`.


I have two question:
- Is your metric a `gauge` metric or `counter` metric?
- What do you get if you use `rate` instead `increase`?

**Comment by itsmylife** [comment]
You can also try;
> stitch something together with a subquery like
> `sum_over_time((increase(foo[1m]) >0)[1h:1m])`
> Slicing up the increases, and discarding drops.
> Choose the shorter interval to match the frequency of underlying data.

cc: @bboreham

**Comment by Maros112358** [comment]
Hello @itsmylife,
my metric is `gauge`. When I used `rate`, I got 
<img width="1470" alt="image" src="https://github.com/grafana/grafana/assets/98359168/0fcfaee7-1c92-4623-a35f-21e5fc0d88fa">

I checked the video, and it looks like I should use `counter` instead of gauge. I didn't know they can be used this way.

**Comment by itsmylife** [comment]
So this is normal in that case then. Is it ok to close this issue?

**Comment by Maros112358** [comment]
I changed the metric type from `gauge` to counter. This is now the underlying data  `foo`
<img width="1479" alt="image" src="https://github.com/grafana/grafana/assets/98359168/7e221b0c-79ad-45d2-9f1b-c301423cd2f1">
This is the `rate(foo[1h])` (besides the spike, it matches perfectly what I need)
<img width="1473" alt="image" src="https://github.com/grafana/grafana/assets/98359168/c0089277-4d7e-477f-8b56-198be8ea5973">
This is the `increase(foo[1h])` 
<img width="1474" alt="image" src="https://github.com/grafana/grafana/assets/98359168/1a5ed1c4-17c0-40b5-9883-a3c0ca7715ae">

**Comment by Maros112358** [comment]
I changed the metric type from `gauge` to `counter`.

This is now the underlying data `foo`
<img width="1475" alt="image" src="https://github.com/grafana/grafana/assets/98359168/1db121cf-a648-42f3-a59e-f77b4427cc16">
This is the `rate(foo[1h])`
<img width="1477" alt="image" src="https://github.com/grafana/grafana/assets/98359168/5993c1c0-5622-401d-9642-70a7e7683eda">

This is the `increase(foo[1h])`
<img width="1482" alt="image" src="https://github.com/grafana/grafana/assets/98359168/86acd535-33e3-41c4-839f-c0ef7d95bda2">

From what I understood in the video and in the docs, the sudden drop (restart) should be accounted for.

**Comment by bboreham** [comment]
`rate` and `increase` should only be used with counters.
However, many (older) Prometheus exporters don't mark which is which, so `rate` and `increase` don't check, and behave the same in every case.

**Comment by Maros112358** [comment]
Hello, I am using Prometheus version `2.3.0`. I haven't managed to find if this version of Prometheus marks which is which (`counter` or `gauge`). I checked that this version came out in 2018, is this too old? 

In the mean time, I will try using query provided by @itsmylife for what I am trying to achieve.

**Comment by bboreham** [comment]
I was talking about exporters, e.g. node-exporter.
No version of Prometheus checks whether you are calling `rate` on a counter.

I advised to use a different feature of Prometheus, subqueries, to compute the function you asked for.
This feature was added in Prometheus v2.7.  The latest version is 2.48.

Suggest if you want to discuss Prometheus features to use the [Prometheus mailing list](https://groups.google.com/g/prometheus-users) or [Github Discussions](https://github.com/prometheus/prometheus/discussions).
